Dangerfile

```diff
@@ -94,9 +94,14 @@ def checkForWeaklyTypedFunctionReturn(line)
   warn("Possibly returning 'any' in a function, prefer having a strongly typed return. `#{file}` at line `#{line}`.") if line =~ /<any>/im
 end
 
-def checkForNpmInstallGlobal(file, line)
+def checkForNpmInstallGlobal(file)
   # Warn developers that they are not supposed to use this flag
-  fail("`npm install` with flag `-g` was found in `#{file}` at `#{line}`. This is not recommended.") if line =~ /npm install -g/
+  fileDiff = git.diff_for_file(file)
+  # Ensure we keep using secure https:// references instead of http://
+  npmInstallMatches = fileDiff.patch.scan(/npm install -g | npm install --global/)
```

Comment:
@eiditakahashi 
This regex is matching both additions and deletions. To match only additions, you'd need to check for the initial `+`. Also, i'm not sure this will match properly since because of the spaces around `|` and the lack of parenthesis to group each of the possible matches. 
In a nutshell, I believe this regex is searching for a single string which satisfies these rules

1. `npm install -g`
2. empty space or empty space
3. `npm install --global`


Consider using something along the lines of `/\+.*(npm install -g)|(npm install --global)/`